### PR TITLE
makes metadata an optional argument from Nulecule file

### PR DESCRIPTION
**Pull request by surajssd**
This closes issue #502

Made `metadata` a keyword argument, hence now it can act as optional argument in `Nulecule` file.

### CODE REVIEW

**Comment by surajssd** [comment]
Need to update test cases as well ;)

**Comment by rtnpro** [comment]
@surajssd The changes look good :+1: Fix the tests, and it will be good to merge :)

File: atomicapp/nulecule/base.py [hunk 1]
```diff
@@ -48,9 +48,9 @@ def __init__(self, id, specversion, metadata, graph, basepath,
         Args:
             id (str): Nulecule application ID
             specversion (str): Nulecule spec version
-            metadata (dict): Nulecule metadata
             graph (list): Nulecule graph of components
             basepath (str): Basepath for Nulecule application
+            metadata (dict): Nulecule metadata
```
Comment (cdrage) [inline atomicapp/nulecule/base.py]: why was this moved?

**Comment by cdrage** [comment]
Tests pass! this LGTM :+1:

**Comment by cdrage** [comment]
Woops.

Sorry guys, I merged this in without checking the commit message. I've since reverted it.

@surajssd Could you please open a new PR with this code and a new commit message to reflect the changes? Perhaps `Make metadata argument optional when building` or something along that line.

**Comment by surajssd** [comment]
@cdrage okay will do that soon!

**Comment by surajssd** [comment]
@cdrage created a new PR #547
